src/main/java/domain/card/Symbol.java

```diff
@@ -1,33 +1,62 @@
 package domain.card;
 
 public enum Symbol {
-    ACE(1, "A"),
-    TWO(2, "2"),
-    THREE(3, "3"),
-    FOUR(4, "4"),
-    FIVE(5, "5"),
-    SIX(6, "6"),
-    SEVEN(7, "7"),
-    EIGHT(8, "8"),
-    NINE(9, "9"),
-    TEN(10, "10"),
-    JACK(10, "J"),
-    QUEEN(10, "Q"),
-    KING(10, "K");
-
-    private int value;
-    private String word;
-
-    Symbol(int value, String word) {
+    ACE(1, "A", new AceCalculator()),
+    TWO(2, "2", (sum) -> 2),
+    THREE(3, "3", (sum) -> 3),
+    FOUR(4, "4", (sum) -> 4),
+    FIVE(5, "5", (sum) -> 5),
+    SIX(6, "6", (sum) -> 6),
+    SEVEN(7, "7", (sum) -> 7),
+    EIGHT(8, "8", (sum) -> 8),
+    NINE(9, "9", (sum) -> 9),
+    TEN(10, "10", (sum) -> 10),
+    JACK(10, "J", (sum) -> 10),
+    QUEEN(10, "Q", (sum) -> 10),
+    KING(10, "K", (sum) -> 10);
+
+    private final int value;
+    private final String pattern;
+    private final SymbolCalculator calculator;
+
+    Symbol(int value, String pattern, SymbolCalculator calculator) {
```

Comment:
```suggestion
    Symbol(int value, String pattern) {
```

ACE를 제외하면 전부 value를 응답하고 있는데요.
그렇다면 위와 같은 생성자를 제공해준다면 불필요한 중복을 제거할 수 있지 않을까요?

Reply:
처음에 그렇게 짰었는데요!
제 생각엔 Symbol을 계산하는 로직을 Symbol이 갖는 것이 좋다고 생각하여 SymbolCalculator를 둔 것이었습니다. 일부 중복이 생기지만, 만약 사용자가 현재의 Symbol들 외에 다른 Symbol을 추가한다고 가정할 때 반드시 Symbol객체를 추가해야 할 것이고, 그 때 valule와 patten외에도 calculator로직을 넣는 것은 큰 작업이 아니라고 생각합니다. 또한 현재는 Ace만 특별한 로직이 있지만 다른 Symbol도 각각 다른 로직을 적용한다고 변경됐을 때 Symbol 객체에 다른 계산 로직을 만들고, 그것을 적용하면 많은 부분을 수정하지 않아도 변경에 대처할 수 있다고 생각합니다.

```
int calculate(int sum, SymbolCalculator calculator) {
        return calculator.calculate(sum);
}
```
위와 같은 방식으로도 전략패턴을 적용할 수 있다고 생각하지만, 이렇게 되면 기존보다 캡슐화가 깨진다고 생각합니다. calculate()함수를 활용하는 클라이언트 쪽에서 전략을 알아야 하기 때문입니다.

SymbolCalculator를 두지 않고도 Symbol이 Symbol의 점수를 계산하는 방식, 아니면 SymbolCalculator를 두지 않아도 계산 로직이라는 전략이 변경됐을 때 쉽게 변경할 수 있는 방식을 알려주시면 감사하겠습니다!

Reply:
잘 이해를 하셨네요 👍 
하지만 제 피드백의 의도가 잘못 전달된 것 같은데요!
SymbolCalculator를 제거하지 않고, 생성자를 2개로 분리해보면 어떨까? 라는 의미의 피드백이였습니다.
코드를 보면 ACE를 제외하면 모든 코드가 value를 응답하고 있는데요.
그렇다면 ACE를 제외하면 매개변수를 3개 받는 생성자는 필요가 없을 것 같아요!

Reply:
수정해봤는데요!
그런데 
enum의 경우 필드를 일종의 상수로 취급하여 final 키워드를 넣어줘야 하니,
새로운 생성자를 다음과 같이 만들어야 했습니다.

```
Symbol(int value, String pattern) {
        this.value = value;
        this.pattern = pattern;
        this.calculator = null;
    }
```

calculate함수는 다음과 같이 짰습니다.

```
int calculate(int sum) {
        if (Objects.isNull(calculator)) {
            return value;
        }
        return calculator.calculate(sum);
    }
```

calculator를 null로 명시하는 것이 조금 어색하기도 했는데요!
그래서 enum의 필드를 final로 선언해주는 관례를 어기는 게 좋을지, 
아니면 지금처럼 생성 시 null을 넣어주는 게 좋을 지 헷갈렸습니다.
어떤 게 낫다고 생각하시는 지 의견 주시면 감사하겠습니다!

Reply:
생각하다보니 이 코드가 생각나서 적용했습니다~!

```
Symbol(int value, String pattern) {
        this.value = value;
        this.pattern = pattern;
        this.calculator = new DefaultSymbolCalculator(pattern);
    }
```


```
private static class DefaultSymbolCalculator implements SymbolCalculator {
        private String pattern;

        private DefaultSymbolCalculator(String pattern) {
            this.pattern = pattern;
        }

        @Override
        public int calculate(int sum) {
            Symbol symbol = findByPattern(pattern);
            return symbol.getValue();
        }
    }

    private static Symbol findByPattern(String pattern) {
        for (Symbol symbol : values()) {
            if (symbol.getPattern().equals(pattern)) {
                return symbol;
            }
        }

        throw new IllegalArgumentException("해당 심볼을 찾을 수 없습니다.");
    }
```

Reply:
리팩토링 잘 해주셨습니다!
추가적으로 말씀드리자면, 인스턴스 변수 초기화는 하나의 생성자에서 해주는 것이 좋습니다.

```java
Symbol(int value, String pattern) {
        this(value, pattern, ignored -> value);
}
```

해당 코드와 어떠한 차이가 있을지 고민해보세요~

Reply:
@jaeyeonling

오... 훨씬 깔끔하네요
일단 코드가 더 깔끔해진다는 장점이 있을 것 같고요.
같은 맥락에서 DefaultSymbolCalculator, findByPattern과 같은 불필요한 클래스/함수를 구현하지 않아도 될 것 같습니다!
또한 보는 입장에서 DefaultSymbolCalculator를 보는 것보다 훨씬 빠르게 로직을 이해할 수 있을 것 같습니다!